build_tools/build_detect_platform

```diff
@@ -444,18 +444,18 @@ if test "$USE_HDFS"; then
   JAVA_LDFLAGS="$JAVA_LDFLAGS $HDFS_LDFLAGS"
 fi
 
-if test "$USE_SSE"; then
-  COMMON_FLAGS="$COMMON_FLAGS -msse4.2"
-elif test -z "$PORTABLE"; then
+if test -z "$PORTABLE" && [ "$TARGET_OS" != AIX ] && [ "$TARGET_OS" != SunOS ]; then
```

Comment:
I think this new condition doesn't do the right thing for AIX/SunOS on ppc64/s390x architectures, if such combinations exist. 
With this new check, you are giving priority to OS types (AIX/SunOS), and excluding setting `POWER` or `COMMON_FLAGS` for them, whereas previously architecture types were given priority. 
So, imo, this line should just be `if test -z "$PORTABLE"; then` and line 454 should go back to `elif [ "$TARGET_OS" != AIX ] && [ "$TARGET_OS" != SunOS ]; then`

Reply:
that's a good point, I wasn't thinking about combinations of OS/CPU.

there is still a problem, though, which is the way you suggested wouldn't give -msse4.2 when PORTABLE=0, USE_SSE=1, and the target os/cpu don't match any of our conditions. anyways i'll figure it out.

Reply:
True! I missed that.